lib/charset.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+// Load modules
+
+var Boom = require('boom');
+var Hoek = require('hoek');
+
+
+// Declare internals
+
+var internals = {};
+
+
+/*
+    https://tools.ietf.org/html/rfc7231#section-5.3.3
+    Accept-Charset: iso-8859-5, unicode-1-1;q=0.8
+*/
+
+exports.charset = function (header, preferences) {
+
+    Hoek.assert(!preferences || Array.isArray(preferences), 'Preferences must be an array');
+    var charsets = header
+        .split(',')
+        .map(internals.getParts)
+        .filter(internals.removeEmpty)
+        .sort(internals.compareByWeight);
+
+    // tack on a default return
```

Comment:
Upper case first character on comments and empty line after to match the hapi code style.